src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -492,6 +528,119 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     }
   }
 
+  /**
+   * Handle the noPlayableRepresentation event, trigger an error if no fallback is possible.
+   * and can trigger event "noPlayableTrack"
+   * @param period - The period that has no playable representation
+   * @param bufferType - The media type that is not playable
+   */
+  private onNoPlayableRepresentation(period: IPeriodMetadata, bufferType: IBufferType) {
+    const periodHasAdaptationForType =
+      period.adaptations[bufferType] !== undefined &&
+      period.adaptations[bufferType].length > 0;
+    const firstPlayableAdaptation = findFirstPlayableAdaptation(period, bufferType);
+
+    if (!periodHasAdaptationForType) {
+      log.debug(
+        `TS: The period does not have adaptation for ${bufferType} there is no track to choose`,
+      );
+      return;
+    }
+
+    if (bufferType === "text") {
+      const textTypeInfo = getPeriodItem(this._storedPeriodInfo, period.id)?.[bufferType];
+      if (textTypeInfo !== undefined) {
+        const storedSettings = null;
+        textTypeInfo.storedSettings = storedSettings;
+        textTypeInfo.dispatcher?.updateTrack(storedSettings);
+      }
+      return;
```

Comment:
By returning early, we don't send the `trackUpdate` event (and maybe `textTrackChange` if it us the current Period?).